aiohttp/client.py

```diff
@@ -149,6 +150,37 @@
 else:
     SSLContext = None
 
+if sys.version_info >= (3, 11) and TYPE_CHECKING:
+    from typing import Unpack
+
+
+class _RequestOptions(TypedDict, total=False):
+    params: Union[Mapping[str, str], None]
+    data: Any
+    json: Any
+    cookies: Union[LooseCookies, None]
+    headers: Union[LooseHeaders, None]
+    skip_auto_headers: Union[Iterable[str], None]
+    auth: Union[BasicAuth, None]
+    allow_redirects: bool
+    max_redirects: int
+    compress: Union[str, None]
```

Comment:
This contradicts the documentation for `compress` as `bool`:

https://github.com/aio-libs/aiohttp/blob/48a5e07ad833bd1a8fcb2ce6f85a41ad0cef9dc6/docs/client_reference.rst#L431-L434

Reply:
It's copied from _request(), not something related to this PR specifically. Feel free to open a PR to fix it. Looking at the code, it's the docs that are inaccurate, though we could expand the typing to allow bool:
https://github.com/aio-libs/aiohttp/blob/48a5e07ad833bd1a8fcb2ce6f85a41ad0cef9dc6/aiohttp/client_reqrep.py#L434-L436

Reply:
So, basically:
None/False/"" = Disable compression
str = use this compression method
Non-str (e.g. True) = use deflate

Reply:
I'm not entirely clear how the str is used though, I think maybe only deflate and gzip are valid..